src/spikeinterface/postprocessing/principal_component.py

```diff
@@ -269,18 +269,13 @@ def get_some_projections(self, channel_ids=None, unit_ids=None, kept_indices=Non
                 sparse_projection, local_chan_inds = self.get_projections_one_unit(
                     unit_id, sparse=True, kept_indices=kept_indices
                 )
-
                 # keep only requested channels
                 channel_mask = np.isin(local_chan_inds, channel_indices)
-                sparse_projection = sparse_projection[:, :, channel_mask]
-                local_chan_inds = local_chan_inds[channel_mask]
+                if sum(channel_mask) != len(channel_mask):
+                    sparse_projection = sparse_projection[:, :, channel_mask]
 
                 spike_mask = np.flatnonzero(spike_unit_indices == unit_index)
-                proj = np.zeros((spike_mask.size, num_components, channel_indices.size), dtype=dtype)
-                # inject in requested channels
-                channel_mask = np.isin(channel_indices, local_chan_inds)
-                proj[:, :, channel_mask] = sparse_projection
```

Comment:
this array was usless and just filling up memory

Reply:
Tests are not passing... We've also worked this morning with @samuelgarcia to refactor some key portions of this PR. We should merge them and then rebase the two PR (merging_units and meta_merging). Let me know how you want to proceed

Reply:
Ok,let me do that because I have some local changes with improved testing for the get_some_waveforms/projections

Reply:
As you want, then go ahaed. Also, I think that in get_some_projections and get_some_waveforms, we should gain some memory by looking at the number of channels in the union of all the sparsity mask, for the unit_ids that we want to select, and only create a buffer with such a size. But I'll double check